plugins/modules/grafana_datasource.py

```diff
@@ -679,7 +687,9 @@ def __init__(self, module):
             self.headers["Authorization"] = "Bearer %s" % module.params['grafana_api_key']
         else:
             self.headers["Authorization"] = basic_auth_header(module.params['url_username'], module.params['url_password'])
-            self.switch_organisation(module.params['org_id'])
+            if module.params['org_name']:
+                module.params['org_id'] = self.organization_by_name(module.params['org_name'])
+            self.switch_organization(module.params['org_id'])
```

Comment:
I'm not a fan of re-declaring this variable. How about we define a new one and use the ternay-operator:

```suggestion
            org_id = self.organization_by_name(module.params['org_name']) if module.params['org_name'] else module.params['org_id']
            self.switch_organization(org_id)
```